src/include/ipc/probe.h

```diff
@@ -0,0 +1,106 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2019 Intel Corporation. All rights reserved.
+ *
+ * Author: Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ *         Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+/**
+ * \file include/ipc/probe.h
+ * \brief Probe IPC definitions
+ * \author Adrian Bonislawski <adrian.bonislawski@linux.intel.com>
+ * \author Artur Kloniecki <arturx.kloniecki@linux.intel.com>
+ */
+
+#ifndef __IPC_PROBE_H__
+#define __IPC_PROBE_H__
+
+#include <ipc/header.h>
+#include <sof/bit.h>
+#include <stdint.h>
+
+#define PROBE_PURPOSE_EXTRACTION	0x1
+#define PROBE_PURPOSE_INJECTION		0x2
+
+#define PROBE_EXTRACT_SYNC_WORD		0xBABEBEBA
+
+#define PROBE_FMT_TYPE_SHIFT		31
+#define PROBE_STANDARD_TYPE_SHIFT	27
+#define PROBE_AUDIO_FMT_SHIFT		23
+#define PROBE_SAMPLE_RATE_SHIFT		19
+#define PROBE_NB_CHANNELS_SHIFT		14
+#define PROBE_SAMPLE_SIZE_SHIFT		12
+#define PROBE_CONTAINER_SIZE_SHIFT	10
+#define PROBE_SAMPLE_FMT_SHIFT		9
+#define PROBE_SAMPLE_END_SHIFT		8
+#define PROBE_INTERLEAVING_ST_SHIFT	7
+
+#define PROBE_FMT_TYPE_MASK		MASK(31, 31)
+#define PROBE_STANDARD_TYPE_MASK	MASK(30, 27)
+#define PROBE_AUDIO_FMT_MASK		MASK(26, 23)
+#define PROBE_SAMPLE_RATE_MASK		MASK(22, 19)
+#define PROBE_NB_CHANNELS_MASK		MASK(18, 14)
+#define PROBE_SAMPLE_SIZE_MASK		MASK(13, 12)
+#define PROBE_CONTAINER_SIZE_MASK	MASK(11, 10)
+#define PROBE_SAMPLE_FMT_MASK		MASK(9, 9)
+#define PROBE_SAMPLE_END_MASK		MASK(8, 8)
+#define PROBE_INTERLEAVING_ST_MASK	MASK(7, 7)
+
```

Comment:
Macros looks a lot neater and easier to read if the prefix remains constant with all the differentiation at the rear. e.g. PROBE_SHIFT_x and PROBE_MASK_x

Reply:
done